File: Moyen/Factorial_vs_Exponential/Factorial_vs_Exponential.js
const K = parseInt(readline());
let tabNumber = [];
let tabFinal = [];
var inputs = readline().split(' ');
for (let i = 0; i < K; i++) {
    const A = parseFloat(inputs[i]);
    tabNumber.push(A);
}

let stringN = "";
for (let i=0; i<tabNumber.length; i++) {
    let Sum = 0;
    let N = 0;
    let SumN = 0;
    while (Math.log(tabNumber[i]) > SumN) {
        Sum += Math.log(N+1);
        SumN = Sum/(N+1);
        N++;
    }
    if (i+1 === tabNumber.length) { stringN += N; }
    else { stringN += N +" "; }
}
console.log(stringN);

File: Moyen/Factorial_vs_Exponential/Factorial_vs_Exponential.py
import sys
import math

k = int(input())
tabNumber = []
tabFinal = []
stringN = ''
for i in input().split():
    tabNumber.append(float(i))
for number in tabNumber:
    sum = 0
    n = 0
    sumN = 0
    while (math.log(number) > sumN):
        sum += math.log(n+1)
        sumN = sum/(n+1)
        n += 1
    if number == tabNumber[-1]:
        stringN += str(n)
    else:
        stringN += str(n) + ' '
print(stringN)
